common/app/conf/HealthCheck.scala

```diff
@@ -81,49 +81,64 @@ private[conf] trait HealthCheckCache extends HealthCheckFetcher {
   protected val cache = AkkaAgent[List[HealthCheckResult]](List[HealthCheckResult]())
   def get() = cache.get()
 
-  def allSuccessful: Boolean = {
-    get() match {
-      case Nil => false
-      case nonEmpty => nonEmpty.forall(_.recentlySucceed)
-    }
-  }
-  def anySuccessful: Boolean = get().exists(_.recentlySucceed)
-
   def fetchPaths(testPort: Int, paths: String*): Future[Unit] = {
     log.info("Fetching HealthChecks...")
     fetchResults(testPort, paths:_*).map(allResults => cache.send(allResults.toList))
   }
-}
 
-trait CachedHealthCheckController extends Controller with Results with ExecutionContexts with Logging {
+}
 
-  val paths: Seq[String]
-  val testPort: Int
+sealed trait Policy
```

Comment:
Good work using Types instead of a boolean for this. Although, `Policy` will be defined on the package here, which is quite a generic name. May be slightly better to call it `HealthCheckPolicy`